File: app/src/main/java/com/islam/retrofitsoap/webservice/response/simplexml/WeatherResponseModel.java
package com.islam.retrofitsoap.webservice.response.simplexml;



import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Namespace;
import org.simpleframework.xml.Root;

import java.util.List;


@Root(name = "getWeatherbyCityNameResponse")
@Namespace(reference = "http://WebXml.com.cn/")
public class WeatherResponseModel {
    @ElementList(name = "getWeatherbyCityNameResult")
    public List<String> result;
}

File: app/src/main/java/com/islam/retrofitsoap/webservice/response/tikxmk/WeatherResponseModel.kt
package com.islam.retrofitsoap.webservice.response.tikxmk

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.TextContent
import com.tickaroo.tikxml.annotation.Xml


@Xml(name = "getWeatherbyCityNameResult")
class WeatherResponseModel {
    @Element(name = "getWeatherbyCityNameResult")
    var result: List<StringWorkaround>? = null
}

@Xml(name = "string")
class StringWorkaround {
    @TextContent
    var string: String? = null
}